tests/erc721/erc721_test.go

```diff
@@ -162,6 +171,289 @@ func TestOpenSeaProxyApproval(t *testing.T) {
 	}
 }
 
+func testApprovalForAllEvents(t *testing.T, filter *ERC721Filterer, start uint64, want []*ERC721ApprovalForAll) {
+	ignore := ethtest.Comparers(cmpopts.IgnoreFields(ERC721ApprovalForAll{}, "Raw"))
+
+	iter, err := filter.FilterApprovalForAll(&bind.FilterOpts{
+		Start:   start + 1,
+		End:     nil,
+		Context: nil,
+	}, nil, nil)
+	if err != nil {
+		t.Fatalf("%T.FilterApprovalForAll(nil, nil, nil) error %v", filter, err)
+	}
+	defer iter.Close()
+
+	var gotEvents []*ERC721ApprovalForAll
+	for iter.Next() {
+		gotEvents = append(gotEvents, iter.Event)
+	}
+
+	if diff := cmp.Diff(want, gotEvents, ignore...); diff != "" {
+		t.Errorf("After %T deployment and single ownership transfer; Transfer events diff (-want +got):\n%s", filter, diff)
+	}
+}
+
+func TestOpenSeaProxyPreApproval(t *testing.T) {
+	sim, _, _ := deploy(t)
+
+	tests := []struct {
+		name            string
+		from            int
+		to              int
+		hasProxy        bool
+		wantPreApproved bool
+		wantEvents      []*ERC721ApprovalForAll
+	}{
+		{
+			name:            "mint without proxy",
+			from:            -1,
+			to:              tokenReceiver,
+			hasProxy:        false,
+			wantPreApproved: false,
+			wantEvents:      nil,
+		},
+		{
+			name:            "mint with proxy",
+			from:            -1,
+			to:              tokenReceiver,
+			hasProxy:        true,
+			wantPreApproved: true,
+			wantEvents: []*ERC721ApprovalForAll{
+				{
+					Owner:    sim.Addr(tokenReceiver),
+					Operator: sim.Addr(proxy),
+					Approved: true,
+				},
+			},
+		},
+		{
+			name:            "transfer without proxy",
+			from:            tokenOwner,
+			to:              tokenReceiver,
+			hasProxy:        false,
+			wantPreApproved: false,
+			wantEvents:      nil,
+		},
+		{
+			name:            "transfer with proxy",
+			from:            tokenOwner,
+			to:              tokenReceiver,
+			hasProxy:        true,
+			wantPreApproved: true,
+			wantEvents: []*ERC721ApprovalForAll{
+				{
+					Owner:    sim.Addr(tokenReceiver),
+					Operator: sim.Addr(proxy),
+					Approved: true,
+				},
+			},
+		},
+		{
+			name:            "burn without proxy",
+			from:            tokenOwner,
+			to:              -1,
+			hasProxy:        false,
+			wantPreApproved: false,
+			wantEvents:      nil,
+		},
+		{
+			name:            "burn with proxy",
+			from:            tokenOwner,
+			to:              -1,
+			hasProxy:        false,
+			wantPreApproved: false,
+			wantEvents:      nil,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			sim, nft, filter := deploy(t)
+			start := sim.BlockNumber().Uint64()
+
+			toAddr := common.HexToAddress("0x0")
+			if tt.to >= 0 {
+				toAddr = sim.Addr(tt.to)
+			}
+
+			if tt.hasProxy {
+				openseatest.SetProxyTB(t, sim, toAddr, sim.Addr(proxy))
+			}
```

Comment:
I left it intentionally outside because the table also allows to set `hasProxy: true` for burns. So to avoid unexpected behavior in case someone decides to do so.